### DISCUSSION TITLE
운영체제 - 가상 메모리

### DISCUSSION THREAD

**Discussion by zini9188** [Shortudy-10th/tech-for-developer#148] [질의 응답]
질의응답 남겨주세요

**Comment by zini9188** [comment]
## Q. 가상 메모리를 사용해서 얻는 이점이 뭔가요?
프로세스가 오류로 인해 비정상 종료되더라도 운영체제가 해당 프로세스의 주소 위치를 알고 있으므로, 자원을 회수할 수 있고 해당 공간을 재사용할 수 있게 됩니다. 또한 독립적인 가상 주소 공간을 제공하여 프로세스 간의 오류나 충돌 없이 동작하도록 해줍니다.

**Reply by zini9188** [reply]
### Q. 가상 메모리의 단점은 무엇인가요?
가상 메모리는 페이지 폴트나 오버헤드로 인한 시스템 성능 저하로 물리 메모리를 사용하는 것보다 속도가 느리다는 단점이 있습니다.

**Comment by seongeun42** [comment]
## Q. 가상 메모리란 무엇인가요?

가상메모리란 물리 메모리의 크기와 상관없이 더 큰 메모리 공간을 사용할 수 있는 것처럼 보이게 하는 메모리 관리 기법 중 하나입니다.

**Reply by seongeun42** [reply]
## Q. 가상메모리는 어떤 방식으로 작동하나요?

메모리 오버레이 기법을 사용합니다. 메모리 오버레이란 전체 프로세스를 메모리에 전부 로드하는 것이 아니라, 가장 중요한 부분과 당장 필요한 부분만 나누어서 로드하고 다른 기능이 필요해지면 기존에 로드한 부분을 스왑 영역으로 보내고 해당 기능이 있는 부분을 다시 적재하는 방식입니다.

**Comment by watty-exclamationMark** [comment]
## Q. 가상메모리 구현방법은 어떤 것이 있나요?
demand paging(요구 페이징), demand segmentation(요구 세그멘테이션), 세그멘테이션-페이징 혼용 기법이 있습니다.

**Comment by watty-exclamationMark** [comment]
## Q. MMU가 필요한 이유는 무엇인가요?
가상 메모리를 사용하기 위해서는 가상 주소와 물리 주소를 둘 다 알고 있어야 하기 때문입니다.
가상 메모리 접근이 필요할 때, 해당 가상 주소를 실제 물리 주소 값으로 변환해주는 하드웨어가 MMU(Memory Management Unit)입니다.

**Comment by sikutisa** [comment]
## Q. Locality of Reference는 무엇인가요?
동일한 값 또는 해당 값에 관계된 스토리지 위치가 자주 액세스되는 특성으로 지역성의 원리(principle of locality)라고도 불립니다.
